src/pages/index.js

```diff
@@ -1,87 +1,237 @@
+  profileForm,
+  cardForm,
+  profileEditButton,
+  addNewCardButton,
+  profileTitleInput,
+  profileDescriptionInput,
+  //deleteCardForm,
+  avatarUpdateForm,
+  avatarImage,
+  addCardModal,
+} from "../utils/constants.js";
+// import avatarImage from "../images/jacques-cousteau.jpg";
+// const avatarImageEl = document.querySelector(".profile__image");
+// avatarImageEl.src = avatarImage;
 // console.log("test");
 
-// const newCardPopup = new PopupWithForm(constants.selectors.addModal, () =>
+// const newCardPopup = new PopupWithForm(addCardModal, () =>
 //   console.log("clicked")
 // );
 // newCardPopup.open();
 
 // newCardPopup.close();
 
-const cardPreview = new PopupWithImage("#preview-image-modal");
-cardPreview.setEventListeners();
+/*Functions*/
 
-function handleImageClick(name, link) {
-  cardPreview.open({ name, link });
+function openPreviewModal(data) {
+  previewImagePopup.open(data);
 }
 
-function createCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", handleImageClick);
-  return cardElement.getView();
+/* Instantiations*/
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "d5d1d874-a161-48bd-b3b4-ae8e38400ed9",
+    "Content-Type": "application/json",
+  },
+});
+
+let cardSection;
+
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo(res);
+    userInfo.setUserAvatar(res);
+  })
+  .catch((err) => {});
+
+api
+  .getInitialCards()
+  .then((res) => {
+    cardSection = new Section(
+      { items: res, renderer: renderCard },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    selectors.cardTemplate,
+    openPreviewModal,
+    handleCardDeleteClick,
+    handleLikeClick
+  ).getCardElement();
+  cardSection.addItem(card);
 }
 
-const userInformation = new UserInfo({
+const userInfo = new UserInfo({
   name: ".profile__title",
   description: ".profile__description",
+  userAvatarSelector: ".profile__avatar",
 });
 
-const profileEditForm = new PopupWithForm("#profile-edit-modal", (data) => {
-  console.log("Form data", data);
-  userInformation.setUserInfo({
-    title: data.title,
-    description: data.description,
+const profileEditForm = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileEditSubmit
+);
+const addCardPopup = new PopupWithForm("#add-card-modal", handleCardFormSubmit);
+const previewImagePopup = new PopupWithImage("#modal__preview-card");
+const confirmDeletePopup = new PopupWithConfirm("#delete-card-modal");
+
+const avatarImagePopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarFormSubmit
+);
+
+/*Validation*/
+
+const editFormValidator = new FormValidator(config, profileForm);
+const addCardValidator = new FormValidator(config, cardForm);
+const avatarValidator = new FormValidator(config, avatarUpdateForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```